packages/models-library/src/models_library/services_types.py

```diff
@@ -68,7 +92,7 @@ def __get_pydantic_core_schema__(
         return core_schema.no_info_after_validator_function(cls, handler(str))
 
     @classmethod
-    def validate(cls, v: "RunID | str", _: ValidationInfo) -> "RunID":
+    def validate(cls, v: "ServiceRunID | str", _: ValidationInfo) -> "ServiceRunID":
```

Comment:
Q: should this not raise a ValueError instead of TypeError?

Reply:
This is OK since it knows how to deal with it